testsuite/integration/ws/src/test/java/org/jboss/as/test/integration/ws/authentication/policy/AuthenticationPolicyContextTestCase.java

```diff
@@ -0,0 +1,323 @@
+        war.addAsResource(tccl.getResource( RESOURCE_DIR + "sts-config.properties"), "sts-config.properties");
+        war.addAsWebInfResource(createFilteredAsset("resources/WEB-INF/wsdl/PicketLinkSTS.wsdl"), "wsdl/PicketLinkSTS.wsdl");
+        war.addAsWebInfResource(tccl.getResource( RESOURCE_DIR + "WEB-INF/beans.xml"), "beans.xml");
+        war.addAsWebInfResource(tccl.getResource( RESOURCE_DIR + "WEB-INF/jboss-web-ws.xml"), "jboss-web-ws.xml");
+        war.addAsWebInfResource(tccl.getResource( RESOURCE_DIR + "WEB-INF/jboss-wsse-server.xml"), "jboss-wsse-server.xml");
+        war.addAsManifestResource(tccl.getResource(RESOURCE_DIR + "META-INF/jboss-deployment-structure.xml"), "jboss-deployment-structure.xml");
+        war.addAsManifestResource(tccl.getResource(RESOURCE_DIR + "META-INF/jboss-webservices.xml"), "jboss-webservices.xml");
+        war.addAsResource(AuthenticationPolicyContextTestCase.class.getPackage(), "dummmy-ws-handler.xml", "org/jboss/as/test/integration/ws/authentication/policy/resources/dummmy-ws-handler.xml");
+        return war;
+    }
+
+    private static StringAsset createFilteredAsset(String resourceName) {
+        return new StringAsset(replaceNodeAddress(resourceName));
+    }
+
+    private static String replaceNodeAddress(String resourceName) {
+        String content = null;
+        try {
+            content = IOUtils.toString(AuthenticationPolicyContextTestCase.class.getResourceAsStream(resourceName), "UTF-8");
+        } catch (IOException e) {
+            throw new RuntimeException("Exception during replacing node address in resource", e);
+        }
+        return content.replaceAll("@node0@", NetworkUtils.formatPossibleIpv6Address(System.getProperty("node0", "127.0.0.1")));
+    }
+
+    @Before
+    public void before() {
+        try {
+            initServerConfiguration(PICKETLINK_STS, PICKETLINK_STS_WS);
+        } catch (Throwable t) {
+            LOGGER.error("Server configuration initialization failed.", t.getMessage(), t);
+            fail("Throwable:" + t.getMessage());
+        }
+    }
+
+    @After
+    public void after() throws IOException {
+        deployer.undeploy(PICKETLINK_STS);
+        deployer.undeploy(PICKETLINK_STS_WS);
+    }
+
+    @BeforeClass
+    public static void initClient() throws Exception {
+        wsClient = new WSTrustClient("PicketLinkSTS", "PicketLinkSTSPort",
+                getHttpUrl(DEFAULT_HOST, DEFAULT_PORT) + "picketlink-sts/PicketLinkSTS", new WSTrustClient.SecurityInfo(USERNAME, PASSWORD));
+    }
+
+    @AfterClass
+    public static void closeClient() throws Exception {
+        if (wsClient != null) {
+            wsClient.close();
+        }
+    }
+
+    private static String getHttpUrl(String host, int port) {
+        return "http://" + host + ":" + port + "/";
+    }
+
+    private static String getHostname() {
+        String hostname = System.getProperty("node0");
```

Comment:
Just use @ArquillianResource URI uri; to inject the server web URI, and get the host from that, there should be no need for this.